firebase-firestore/src/main/java/com/google/firebase/firestore/auth/FirebaseAuthCredentialsProvider.java

```diff
@@ -60,30 +64,47 @@
   private boolean forceRefresh;
 
   /** Creates a new FirebaseAuthCredentialsProvider. */
-  public FirebaseAuthCredentialsProvider(InternalAuthProvider authProvider) {
-    this.authProvider = authProvider;
+  public FirebaseAuthCredentialsProvider(Deferred<InternalAuthProvider> deferredAuthProvider) {
     this.idTokenListener =
         token -> {
           synchronized (this) {
-            currentUser = getUser();
+            currentUser = getUser(internalAuthProviderProviderRef.get());
             tokenCounter++;
 
             if (changeListener != null) {
               changeListener.onValue(currentUser);
             }
           }
         };
-    currentUser = getUser();
+
+    internalAuthProviderProviderRef = new AtomicReference<>(() -> null);
+    currentUser = getUser(internalAuthProviderProviderRef.get());
     tokenCounter = 0;
 
-    authProvider.addIdTokenListener(idTokenListener);
+    deferredAuthProvider.whenAvailable(
+        provider -> {
+          synchronized (this) {
+            internalAuthProviderProviderRef.set(provider);
+            currentUser = getUser(internalAuthProviderProviderRef.get());
+            if (changeListener != null) {
+              changeListener.onValue(currentUser);
+            }
+          }
+          provider.get().addIdTokenListener(idTokenListener);
+        });
   }
 
   @Override
   public synchronized Task<String> getToken() {
     boolean doForceRefresh = forceRefresh;
     forceRefresh = false;
-    Task<GetTokenResult> res = authProvider.getAccessToken(doForceRefresh);
+
```

Comment:
What do you mean by "inline"?

Reply:
You can ignore this since this is existing code, but I can't figure out why we need this extra variable. By inlining I mean replace "doForceRefresh" with "forceRefresh" (and setting "forceRefresh" to false a bit later).

Reply:
I can't figure it out either. The only difference it would make would be if `internalAuthProvider.getAccessToken()` throws an exception, which seems like a pretty extreme case against which I doubt we need to guard. I've implemented this "inlining" of `doForceRefresh`.